From f58aa2cfa49c226f7233995e5359954512ee97c0 Mon Sep 17 00:00:00 2001
Message-ID: <f58aa2cfa49c226f7233995e5359954512ee97c0.1741959406.git.sam@gentoo.org>
From: Sam James <sam@gentoo.org>
Date: Fri, 14 Mar 2025 13:36:44 +0000
Subject: [PATCH] PR119287

---
 gcc/match.pd | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gcc/match.pd b/gcc/match.pd
index 89612d1b15b6..98f637373242 100644
--- a/gcc/match.pd
+++ b/gcc/match.pd
@@ -5284,7 +5284,7 @@ DEFINE_INT_AND_FLOAT_ROUND_FN (RINT)
 	 || tree_int_cst_sgn (@2) >= 0)
 	&& wi::to_wide (@3) == wi::set_bit_in_zero (shift, prec))
     (with { auto mask = wide_int::from (wi::to_wide (@2), prec, UNSIGNED); }
-     (bit_and @0 { wide_int_to_tree (type, mask << shift); }))))))
+     (bit_and (convert @0) { wide_int_to_tree (type, mask << shift); }))))))
 
 /* ~(~X >> Y) -> X >> Y (for arithmetic shift).  */
 (simplify

base-commit: 5abe571e0276fafcc6eed27c27abb28943e67c6f
-- 
2.48.1

