https://gcc.gnu.org/bugzilla/show_bug.cgi?id=117047#c36
--- a/gcc/ggc.h
+++ b/gcc/ggc.h
@@ -127,8 +127,7 @@ extern void gt_pch_save (FILE *f);
 
 /* The internal primitive.  */
 extern void *ggc_internal_alloc (size_t, void (*)(void *), size_t,
-				 size_t CXX_MEM_STAT_INFO)
-     ATTRIBUTE_MALLOC;
+				 size_t CXX_MEM_STAT_INFO);
 
 inline void *
 ggc_internal_alloc (size_t s CXX_MEM_STAT_INFO)
@@ -141,7 +140,7 @@ extern size_t ggc_round_alloc_size (size_t requested_size);
 /* Allocates cleared memory.  */
 extern void *ggc_internal_cleared_alloc (size_t, void (*)(void *),
 					 size_t, size_t
-					 CXX_MEM_STAT_INFO) ATTRIBUTE_MALLOC;
+					 CXX_MEM_STAT_INFO);
 
 inline void *
 ggc_internal_cleared_alloc (size_t s CXX_MEM_STAT_INFO)
